### ISSUE TITLE
feat[train]: Implement learning rate scheduling

### ISSUE THREAD

**Issue by Hamza-Bin-Aamir** [Hamza-Bin-Aamir/PIVOT#30]
Warmup phase: 5 epochs (0 → 3e-4)
Cosine annealing: 95 epochs (3e-4 → 1e-6)
Support manual scheduling adjustments
